resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientDistributedCacheManager.scala

```diff
@@ -68,10 +68,10 @@ private[spark] class ClientDistributedCacheManager() extends Logging {
       link: String,
       statCache: Map[URI, FileStatus],
       appMasterOnly: Boolean = false): Unit = {
-    val destStatus = statCache.getOrElse(destPath.toUri(), fs.getFileStatus(destPath))
+    val destStatus = statCache.getOrElseUpdate(destPath.toUri(), fs.getFileStatus(destPath))
```

Comment:
Cant this not be `getFileStatus(fs, destPath.toUri(), statCache)` ?
That will help with [this comment](https://github.com/apache/spark/pull/41821/files#r1264346051).